File: C_Sharp_code/Example/WindowsFormsApp/VB_demo23/Form1.vb
Imports System.IO

Public Class Form1
    Private Sub Form1_Load(sender As Object, e As EventArgs) Handles MyBase.Load

    End Sub
    Private Sub button1_Click(sender As Object, e As EventArgs) Handles button1.Click
        'imports IO類別
        '變數名稱與類別名稱不區分大小寫 命名相同會呼叫不出方法
        Dim fs As FileStream = File.Create("text1.txt")
        Dim bytes() As Byte = {74, 65, 86, 65}
        fs.Write(bytes, 0, bytes.Length)
        fs.Close()
        MessageBox.Show("fils saved")
    End Sub

    Private Sub button2_Click(sender As Object, e As EventArgs) Handles button2.Click
        File.Copy("text1.txt", "text1_new.txt")
        If File.Exists("text1_new.txt") Then
            MessageBox.Show("text1_new.txt Exists")
        Else
            MessageBox.Show("text1_new.txt Not Exists")
        End If
    End Sub

    Private Sub button3_Click(sender As Object, e As EventArgs) Handles button3.Click
        If File.Exists("text1.txt") Then
            MessageBox.Show("text1_new.txt Exists")
        Else
            MessageBox.Show("text1_new.txt Not Exists")
        End If
    End Sub

    Private Sub button4_Click(sender As Object, e As EventArgs) Handles button4.Click
        File.Delete("text1_new.txt")
        If File.Exists("text1_new.txt") Then
            MessageBox.Show("text1_new.txt Exists")
        Else
            MessageBox.Show("text1_new.txt Not Exists")
        End If
    End Sub

    Private Sub button5_Click(sender As Object, e As EventArgs) Handles button5.Click
        Dim f As New FileInfo("text1.txt")
        Dim fs As FileStream = f.Open(FileMode.Open)

        Dim bytes(f.Length) As Byte
        ' Dim bytes As Byte()
        ' ReDim bytes(f.Length)


        Dim c As Integer = Convert.ToInt32(f.Length)

        fs.Read(bytes, 0, c)
        fs.Close()
        For Each t As Byte In bytes
            Console.Write(Convert.ToChar(t))
        Next
        Console.WriteLine()
    End Sub

    Private Sub button6_Click(sender As Object, e As EventArgs) Handles button6.Click
        Dim f As New FileInfo("text2.txt")
        Dim fs As FileStream = f.Create()
        Dim bytes As Byte() = {74, 65, 86, 65}
        fs.Write(bytes, 0, bytes.Length)
        fs.Close()
        f.CopyTo("text2_new.txt", True)
        MessageBox.Show("file copy")
    End Sub

    Private Sub button7_Click(sender As Object, e As EventArgs) Handles button7.Click
        Dim f As New FileInfo("text2_new.txt")
        f.Delete()
        If f.Exists() Then
            MessageBox.Show("text2_new.txt Exists")
        Else
            MessageBox.Show("text2_new.txt Not Exists")
        End If
    End Sub

    Private Sub button8_Click(sender As Object, e As EventArgs) Handles button8.Click
        Dim f As New FileInfo("text1.txt")
        '一個char占用一個byte
        Console.WriteLine("目前檔案(byte):" & f.Length) '取得目前檔案的大小，以位元組為單位。
    End Sub
End Class

File: C_Sharp_code/Example/WindowsFormsApp/CS_demo22/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Forms;
using System.IO;

namespace CS_demo22
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }

        private void button1_Click(object sender, EventArgs e)
        {
            byte[] str = { 74, 65, 86, 65 };
            //插入IO類別
            //File是一種靜態class
            //檔案會產生在 CS_demo22\bin\Debug
            //在工具 -> 方案總管 -> 專案按顯示所有檔案可以看到
            FileStream fs = File.Create("text1.txt");
            fs.Write(str, 0, str.Length);
            fs.Close();
            MessageBox.Show("file saved");
        }

        private void button2_Click(object sender, EventArgs e)
        {
            File.Copy("text1.txt", "text1_new.txt");
            if (File.Exists("text1_new.txt"))
                MessageBox.Show("text1_new.txt Exists");
            else
                MessageBox.Show("text1_new.txt Not Exists");
        }

        private void button3_Click(object sender, EventArgs e)
        {
            if (File.Exists("text1_new.txt"))
                MessageBox.Show("text1_new.txt Exists");
            else
                MessageBox.Show("text1_new.txt Not Exists");
        }

        private void button4_Click(object sender, EventArgs e)
        {
            File.Delete("text1_new.txt");
            if (File.Exists("text1_new.txt"))
                MessageBox.Show("text1_new.txt Exists");
            else
                MessageBox.Show("text1_new.txt Not Exists");
        }

        private void button5_Click(object sender, EventArgs e)
        {
            FileInfo fileInfo = new FileInfo("text1.txt");
            FileStream fs = fileInfo.Open(FileMode.Open);//選擇開啟方法並產生串流物件
            byte[] bytes = new byte[fileInfo.Length];//建立byte陣列空間是text1.txt的長度大小來存放fs串流
            int size = -1;
            fs.Read(bytes, 0, Convert.ToInt32(fileInfo.Length));//將串流來源讀到陣列,從陣列中第0個開始讀到指定的位元組數
            foreach (byte t in bytes)
                Console.Write("{0}", Convert.ToChar(t));
            fs.Close();
            Console.WriteLine("");
        }

        private void button6_Click(object sender, EventArgs e)
        {
            FileInfo fileInfo = new FileInfo("text2.txt");
            FileStream fs = fileInfo.Create();
            byte[] bytes = { 74, 65, 86, 65 };
            fs.Write(bytes,0,bytes.Length);
            fs.Close();
            fileInfo.CopyTo("text2_new.txt",true);//若允許覆寫現有檔案，則為 true，否則為 false
            MessageBox.Show("檔案Copy");
        }

        private void button7_Click(object sender, EventArgs e)
        {
            FileInfo fileInfo = new FileInfo("text2_new.txt");
            fileInfo.Delete();
            if (fileInfo.Exists)
                MessageBox.Show("text2_new.txt Exists");
            else
                MessageBox.Show("text2_new.txt Not Exists");

        }

        private void button8_Click(object sender, EventArgs e)
        {
            FileInfo fileInfo = new FileInfo("text1.txt");
            //一個char占用一個byte
            Console.WriteLine("目前檔案(byte):" + fileInfo.Length); //取得目前檔案的大小，以位元組為單位。
        }
    }
}
